gluten-core/src/main/java/org/apache/gluten/substrait/expression/WindowFunctionNode.java

```diff
@@ -80,20 +91,50 @@ private Expression.WindowFunction.Bound.Builder setBound(
         builder.setUnboundedFollowing(followingBuilder.build());
         break;
       default:
-        try {
-          Long offset = Long.valueOf(boundType);
-          if (offset < 0) {
-            Expression.WindowFunction.Bound.Preceding.Builder offsetPrecedingBuilder =
-                Expression.WindowFunction.Bound.Preceding.newBuilder();
-            offsetPrecedingBuilder.setOffset(0 - offset);
-            builder.setPreceding(offsetPrecedingBuilder.build());
-          } else {
-            Expression.WindowFunction.Bound.Following.Builder offsetFollowingBuilder =
-                Expression.WindowFunction.Bound.Following.newBuilder();
-            offsetFollowingBuilder.setOffset(offset);
-            builder.setFollowing(offsetFollowingBuilder.build());
+        if (boundType instanceof PreComputeRangeFrameBound) {
+          ExpressionNode refNode =
+              ExpressionConverter.replaceWithExpressionTransformer(
+                      ((PreComputeRangeFrameBound) boundType).child().toAttribute(),
+                      JavaConverters.asScalaIteratorConverter(originalInputAttributes.iterator())
+                          .asScala()
+                          .toSeq())
+                  .doTransform(new HashMap<String, Long>());
+          try {
+            Long offset = Long.valueOf(boundType.eval(null).toString());
+            if (offset < 0) {
+              Expression.WindowFunction.Bound.Preceding.Builder refPrecedingBuilder =
+                  Expression.WindowFunction.Bound.Preceding.newBuilder();
+              refPrecedingBuilder.setRef(refNode.toProtobuf());
+              builder.setPreceding(refPrecedingBuilder.build());
+            } else {
+              Expression.WindowFunction.Bound.Following.Builder refFollowingBuilder =
+                  Expression.WindowFunction.Bound.Following.newBuilder();
+              refFollowingBuilder.setRef(refNode.toProtobuf());
+              builder.setFollowing(refFollowingBuilder.build());
+            }
+          } catch (NumberFormatException e) {
+            throw new UnsupportedOperationException(
+                "Unsupported Window Function Frame Type:" + boundType);
```

Comment:
Do we need to add statements to catch this exception? I think Spark parser has already checked the validity of foldable bound (e.g., 100a is not valid). The catching looks unnecessary.